python/ray/serve/_private/application_state.py

```diff
@@ -705,20 +707,23 @@ def _reconcile_build_app_task(self) -> Tuple[Optional[Dict], BuildAppStatus, str
     def _check_routes(
         self, deployment_infos: Dict[str, DeploymentInfo]
     ) -> Tuple[str, str]:
```

Comment:
I know that the old implementation abused this function, but let's extract out the logic for checking duplicate ASGI apps outside of this function.